pandas/core/ops.py

```diff
@@ -1435,6 +1436,82 @@ def add_flex_arithmetic_methods(cls):
     add_methods(cls, new_methods=new_methods)
 
 
+# ----------------------------------------------------------------------
+# Unary
+
+def _unary_method(cls, op, special=True):
+    op_name = _get_op_name(op, special)
+
+    def f(self):
+        if self.ndim > 2:
+            with np.errstate(all='ignore'):
+                # apply op at once based on current impl
+                new_data = op(self._values)
+            result = self._constructor(new_data, self.items,
+                                       self.major_axis, self.minor_axis)
+            return result.__finalize__(self)
+        elif self.ndim == 2:
+            new_data = {i: op(self.iloc[:, i])
+                        for i in range(len(self.columns))}
+            result = self._constructor(new_data, index=self.index, copy=False)
+            # Pin columns instead of passing to constructor for compat with
+            # non-unique columns case
+            result.columns = self.columns
+            return result.__finalize__(self)
+        else:
+            with np.errstate(all='ignore'):
+                new_data = op(self._values)
+            return self._constructor(new_data, name=self.name,
+                                     index=self.index).__finalize__(self)
+
+    f.__name__ = op_name
+
+    return f
+
+
+def pos(values):
+    if is_datetime64_any_dtype(values):
```

Comment:
Why is this done here, instead of on `DatetimeArray`?

Reply:
I think we need a decision about this. To move `__pos__` method to `PandasArray`,  other array dtypes also need it to keep compat with `NumPy` array.

```
>>> +np.array([np.timedelta64(1)])
# array([1], dtype=timedelta64)

>>> +np.array(['a'], dtype=object)
# array(['a'], dtype=object)
```

I think prior choice is simpler:
- Make global `pos` function for all dtypes, and raise if input is `datetime64` (current impl).
- Add `__pos__` method to all `PandasArray`.

Reply:
By `PandasArray`, do you mean ExtensionArray?

On master, `+PandasArray` seems to work

```python
In [21]: +pd.Series([1, 2]).array
Out[21]:
<PandasArray>
[1, 2]
Length: 2, dtype: int64
```

I assume this is via `__array_ufunc__`.

For now, I would prefer that `DatetimeArray` defines a `__pos__` that raises a ValueError or TypeError.